verification/verify.go

```diff
@@ -98,12 +105,19 @@ func NewSegmentVerifier(p *Policy) *SegmentVerifier {
 }
 
 func (sv *SegmentVerifier) Verify(params *Params) (*Params, error) {
+	return sv.verify(verifySignature, params)
+}
+
+type sigVerifyFunc func(params *Params) error
```

Comment:
Perhaps it could be easier for testing purposes if the signature of this function was:

```
type sigVerifyFunc func(addr ethcommon.Address, msg, sig []byte) bool
```

Then, the stub implementation of this type could just be configured to return true or false and we wouldn't need to deal with the test data setup that is currently necessary for generating a valid signature.

Reply:
resolved by d3e4afb5f4f5f4a90b0982d7e18ad55a5af06db2